### ISSUE TITLE
Update DataBlockGroup to avoid workarounds for VRTs to work

### ISSUE THREAD

**Issue by benbowler** [google/site-kit-wp#9419]
## Feature Description

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* [ ] 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by zutigrm** [comment]
Thanks @benbowler 

> New delays should not be added to VRTs.

Not sure if this point is necessary, I see we are not using delays for the stories that were failing/flaky. Did you want to ensure new delay is not added as part of the solution? If we modify the component to resize fs in more reliable way, we probably won't need to add delay anyway

**Comment by benbowler** [comment]
@zutigrm yes I suppose this was to make sure we don't end up back where we started with this refactor as stories that rely on a delay are usually waiting for some render logic that can take an indeterminant amount of time and therefore case the story to fail if a GH Action runner is running slowely.

**Comment by zutigrm** [comment]
@benbowler Thanks, makes sense. AC ✅

**Comment by eugene-manuilov** [comment]
IB ✔

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- Tested the [storybook stories](https://storage.googleapis.com/site-kit-storybook/develop/index.html?path=/story/modules-analytics4-widgets-dashboardoverallpagemetricswidgetga4--ready) and components within the app, confirming that the metrics text in each datablock never overflows each datablock or wraps onto another line as the page is resized. 
  - DashboardOverallPageMetricsWidgetGA4
  - WPDashboardWidgets
- Tested on desktop and various mobile and tablet viewports

https://github.com/user-attachments/assets/1eca4792-08ad-4f90-94f3-d807c66beec8
